pennylane/transforms/decompositions/clifford_t_transform.py

```diff
@@ -401,54 +401,13 @@ def circuit(x, y):
                 if op.num_wires == 1:
                     if op.name in basis_set:
                         d_ops = _rot_decompose(op)
-                    else:
-                        d_ops, g_op = _one_qubit_decompose(op)
-                        gphase_ops.append(g_op)
-                    decomp_ops.extend(d_ops)
+                        decomp_ops.extend(d_ops)
 
-                # Two qubit unitary decomposition with SU(4) rotations
-                elif op.num_wires == 2:
-                    d_ops = _two_qubit_decompose(op)
-                    decomp_ops.extend(d_ops)
-
-                # For special multi-qubit gates and ones constructed from matrix
+                # If we don't know how to decompose the operation
                 else:
-                    try:
-                        # Attempt decomposing the operation
-                        md_ops = op.decomposition()
-                        idx = 0  # might not be fast but at least is not recursive
-                        while idx < len(md_ops):
-                            md_op = md_ops[idx]
-                            if md_op.name not in basis_set or not check_clifford_t(md_op):
-                                # For the gates acting on one qubit
-                                if len(md_op.wires) == 1:
-                                    if md_op.name in basis_set:  # For known recipe
-                                        d_ops = _rot_decompose(md_op)
-                                    else:  # Resort to decomposing manually
-                                        d_ops, g_op = _one_qubit_decompose(md_op)
-                                        gphase_ops.append(g_op)
-
-                                # For the gates acting on two qubits
-                                elif len(md_op.wires) == 2:
-                                    # Resort to decomposing manually
-                                    d_ops = _two_qubit_decompose(md_op)
-
-                                # Final resort (should not enter in an ideal situation)
-                                else:
-                                    d_ops = md_op.decomposition()
-
-                                # Expand the list and iterate over
-                                del md_ops[idx]
-                                md_ops[idx:idx] = d_ops
-                            idx += 1
-
-                        decomp_ops.extend(md_ops)
-
-                    # If we don't know how to decompose the operation
-                    except Exception as exc:
-                        raise ValueError(
-                            f"Cannot unroll {op} into the Clifford+T basis as no rule exists for its decomposition"
-                        ) from exc
+                    raise ValueError(
+                        f"Cannot unroll {op} into the Clifford+T basis as no rule exists for its decomposition"
+                    )
 
```

Comment:
This entire branch can be pruned as once again the `compile` step will have compiled the circuit to the basis set or thrown an error so this branch is not accessible.